frame/dapps-staking/src/tests.rs

```diff
@@ -1264,6 +1264,50 @@ fn new_era_forcing() {
     })
 }
 
+#[test]
+fn new_era_length_is_always_blocks_per_era() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+        let blocks_per_era = mock::BLOCKS_PER_ERA;
+
+        // start new era era and record era number and height
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+        let mut expected_era = mock::DappsStaking::current_era();
+        let mut previous_era_starting_block_height;
```

Comment:
I think you should avoid such empty declarations - instead declare the variable when you need it.
And please remove `height` from name, it's very confusing :D 

Also, if you need to change the value of the variable multiple times, I'd suggest you just re-define it.
E.g.
```
let x = 5;
let x = 7;
```

Reply:
Sorry for the confusion! Perhaps I'm using it wrong. I had a ways heard of 'block height' defined as: "The block height of a particular block is defined as the number of blocks preceding it in the blockchain".

I'll change it to 'block_number".

Reply:
I didn't know that!

I did a check in substrate repo and block height is mentioned a few times.
Block number is mentioned much more though.

For our repo, I suggest we stick with block number since we use that term already in the code (we don't use block height).

Thanks for the explanation though, I learned something :D